### PR TITLE
copy MikeP draft

### PR DESCRIPTION
**Pull request by Sakurann**

### CODE REVIEW

File: draft-terbu-sd-jwt-vc.md [hunk 1]
```diff
@@ -1,22 +1,20 @@
 %%%
-title = "SD-JWT-based Verifiable Credentials with JSON payloads (SD-JWT VC)"
-abbrev = "SD-JWT VC"
-ipr = "trust200902"
-workgroup = "TODO Working Group"
-keyword = ["security", "openid", "sd-jwt"]
+title = "Native JWT Representation of Verifiable Credentials"
```
Comment (tlodderstedt) [inline draft-terbu-sd-jwt-vc.md:2]:
```suggestion
title = "JWT representation of Verifiable Credentials"
```

I don't like "native". What does this mean? I would rather like to have a title along the lines of "Verifiable Credentials with JSON payload following JWT rules with optional selective disclosure support" but I know that sounds pretty artificial. So removing native might be ok.

File: draft-terbu-sd-jwt-vc.md [hunk 2]
```diff
@@ -1,22 +1,20 @@
...
+abbrev = "jwt-vcs"
+ipr= "trust200902"
+area = "Internet"
+workgroup = "None"
+submissiontype = "IETF"
+keyword = ["JOSE","COSE","JWT","CWT"]
```
Comment (tlodderstedt) [inline draft-terbu-sd-jwt-vc.md:8]:
```suggestion
keyword = ["OAUTH","SD-JWT","JWT"]
```

File: draft-terbu-sd-jwt-vc.md [hunk 3]
```diff
@@ -1,22 +1,20 @@
...
 
 [seriesInfo]
 name = "Internet-Draft"
-value = "draft-terbu-sd-jwt-vc-latest"
+value = "draft-prorock-oauth-jwt-vcs-latest"
```
Comment (tlodderstedt) [inline draft-terbu-sd-jwt-vc.md:12]: I thought we wanted to integrate the JWT part into Oliver's draft.

File: draft-terbu-sd-jwt-vc.md [hunk 4]
```diff
@@ -71,88 +139,117 @@ Credentials are cryptographically signed statements about a Subject, typically t
...
 
-The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
-"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this
-document are to be interpreted as described in RFC 2119 [@!RFC2119].
+## Overview
 
-## Terms and Definitions
+This section provides guidance on how to use JSON [@!RFC8259] claimsets
+with JWT [@!RFC7519] registered claims to construct a JWT that can be
+mapped to a verifiable credential. This section also describes how to
```
Comment (tlodderstedt) [inline draft-terbu-sd-jwt-vc.md:197]: what does "mapping" mean?

File: draft-terbu-sd-jwt-vc.md [hunk 5]
```diff
@@ -71,88 +139,117 @@ Credentials are cryptographically signed statements about a Subject, typically t
...
-This specification uses the terms "Holder", "Issuer", "Verifier", defined by
-[@!I-D.ietf-oauth-selective-disclosure-jwt].
+This representation relies on claims registered in the [IANA JSON Web
+Token Claims
+Registry](https://www.iana.org/assignments/jwt/jwt.xhtml#claims)
+whenever possible.
 
-Verifiable Credential (VC):
-:  An Issuer-signed assertion with claims about a Subject.
+Implementers using this representation SHOULD NOT use `vc+ld+json` as an
```
Comment (tlodderstedt) [inline draft-terbu-sd-jwt-vc.md:206]: sounds like a self fulfilling prophecy - why is that needed? I would rather state what the expected `typ` value is.

File: draft-terbu-sd-jwt-vc.md [hunk 6]
```diff
@@ -71,88 +139,117 @@ Credentials are cryptographically signed statements about a Subject, typically t
...
-SD-JWT-based Verifiable Credential (SD-JWT VC):
-: A Verifiable Credential encoded using the Issuance format defined in
-[@!I-D.ietf-oauth-selective-disclosure-jwt].
+### Credential Header
 
-Unsecured payload of an SD-JWT VC:
-: A JSON object containing all selectively disclosable and non-selectively disclosable claims
-of the SD-JWT VC. The unsecured payload acts as the input JSON object to issue
-an SD-JWT VC complying to this specification.
+`typ` MUST use the media type `vc+jwt`.
```
Comment (tlodderstedt) [inline draft-terbu-sd-jwt-vc.md:211]: or `vc+sd-jwt`

File: draft-terbu-sd-jwt-vc.md [hunk 7]
```diff
@@ -71,88 +139,117 @@ Credentials are cryptographically signed statements about a Subject, typically t
...
+  "kid": "https://example.edu/issuers/14#key-0",
+  "alg": "ES256",
+  "typ": "vc+jwt"
+}
+```
 
-* This specification defines
-  - Data model and media types for Verifiable Credentials based on SD-JWTs.
-  - Validation and processing rules for Verifiers and Holders.
+### Credential
```
Comment (tlodderstedt) [inline draft-terbu-sd-jwt-vc.md:223]: no definition of the MUST claims in a JWT VC?

File: draft-terbu-sd-jwt-vc.md [hunk 8]
```diff
@@ -71,88 +139,117 @@ Credentials are cryptographically signed statements about a Subject, typically t
...
+  "iss": "https://example.edu/issuers/14",
+  "sub": "1234567890",
+  "name": "John Doe",
+  "iat": 1516239022,
+  "urn:example:claim": true
+}
+```
 
-# Use Cases
+NOTE: The `vc` and `vp` claims MUST NOT be present when the content
```
Comment (tlodderstedt) [inline draft-terbu-sd-jwt-vc.md:237]: which vc and vp claims? seems this is another self fulfilling prophecy

File: draft-terbu-sd-jwt-vc.md [hunk 9]
```diff
@@ -71,88 +139,117 @@ Credentials are cryptographically signed statements about a Subject, typically t
...
+type header parameter is set to `credential-claims-set+json`.
```
Comment (tlodderstedt) [inline draft-terbu-sd-jwt-vc.md:238]: typ or cty? I don't understand this sentence. Given line 211 states `typ` is `vc+jwt`.

**Review by tlodderstedt — COMMENTED** [review]
I think the joint spec requires several sections explaining the concepts used by both representations, namely the structure of the payload (all the JWT claims), JWT Issuer Metadata, Key Binding, and individual section covering the differences, namely `typ` values and none selectively disclosable claims (for SD-JWT).

**Comment by awoie** [comment]
Closing this PR since PR #131 was merged.
